Assets/MRTK/SDK/Experimental/Elastic/Scripts/QuaternionElasticSystem.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Runtime.CompilerServices;
+using UnityEngine;
+
+[assembly: InternalsVisibleTo("Microsoft.MixedReality.Toolkit.Tests.PlayModeTests")]
+namespace Microsoft.MixedReality.Toolkit.Experimental.Physics
+{
+    public class QuaternionElasticSystem : ElasticSystem<Quaternion>
+    {
+        private Quaternion currentValue;
+        private Quaternion currentVelocity;
+
+        private QuaternionElasticExtent extent;
+        private ElasticProperties elasticProperties;
+
+        public QuaternionElasticSystem(Quaternion initialValue,
+                                       Quaternion initialVelocity,
+                                       QuaternionElasticExtent extentInfo,
+                                       ElasticProperties elasticProperties)
+        {
+            currentValue = initialValue;
+            currentVelocity = initialVelocity;
+            this.extent = extentInfo;
+            this.elasticProperties = elasticProperties;
+        }
+
+        /// <inheritdoc/>
+        public override Quaternion ComputeIteration(Quaternion forcingValue, float deltaTime)
```

Comment:
Can you add some comments for the internals of this function for roughly the intent of each of large sections within here?

Reply:
Definitely, working on that now